src/libraries/System.IO.IsolatedStorage/src/System.IO.IsolatedStorage.csproj

```diff
@@ -10,6 +10,10 @@
   <ItemGroup Condition="'$(TargetPlatformIdentifier)' != ''">
     <Compile Include="System\IO\IsolatedStorage\IsolatedStorageException.cs" />
     <Compile Include="System\IO\IsolatedStorage\IsolatedStorageFile.cs" />
+    <Compile Condition="'$(TargetPlatformIdentifier)' == 'Android' or '$(TargetPlatformIdentifier)' == 'iOS' or '$(TargetPlatformIdentifier)' == 'MacCatalyst' or '$(TargetPlatformIdentifier)' == 'tvOS'"
+             Include="System\IO\IsolatedStorage\IsolatedStorageFile.AnyMobile.cs" />
+    <Compile Condition="'$(TargetPlatformIdentifier)' != 'Android' and '$(TargetPlatformIdentifier)' != 'iOS' and '$(TargetPlatformIdentifier)' != 'MacCatalyst' and '$(TargetPlatformIdentifier)' != 'tvOS'"
```

Comment:
As mentioned above, `TargetsMobile` shouldn't be used for building libraries. It can be used when conditioning based on the TargetOS, i.e. to not run tests.